classes/submission/reviewAssignment/Repository.php

```diff
@@ -227,6 +227,39 @@ public function deleteMany(Collector $collector)
         }
     }
 
+    /**
+     * Delete all review assignments for a given context ID.
+     *
+     * @param int $contextId
+     * @return void
+     */
+    public function deleteByContextId($contextId)
+    {
+        // using reviewAssignmentCollector to fetch and delete all review assignments for the context
+        $reviewAssignmentCollector = $this->getCollector();
+        $reviewAssignmentCollector->filterByContextIds([$contextId]);
+        $reviewAssignments = $reviewAssignmentCollector->getMany();
+    
+        foreach ($reviewAssignments as $reviewAssignment) {
+            $this->delete($reviewAssignment);
+        }
+    
+        // delete review rounds associated with this context
+        $this->deleteReviewRoundsByContextId($contextId);
```

Comment:
Best to just move the `getDAO` / `deleteByContextId` calls directly here, rather than creating a new function.